0017-negotiation-and-execution-protocol.adoc

```diff
@@ -0,0 +1,544 @@
+
+We could specify several problems:
+
+. The Accept/Decline feels weird, it often does not make sense in specific use-case context
+. We do not clearly distinguish trade negotiation from execution at the moment.
+. It is hard to add the concept of orders and an order-book to COMIT.
+
+This spike includes:
+
+. Solution proposal for splitting trade negotiation and execution in different protocols.
+. Solution proposal for the "How to update an offer"/"Offers should not get stale" problem.
+. Defining the base for a first negotiation protocol.
+. Defining the changes needed to the existing execution protocol RFC003.
+
+== Research ==
+
+=== Constraints
+
+Before defining the terminology certain constraints shall be defined, so we know what is already regarded as given.
+
+==== Protocol Constraints
+
+As we build upon libp2p for messaging all protocols should adhere to the libp2p specification. The proposed first negotiation protocol will be a new libp2p protocol.
+
+For the sake of simplicity we assume that there is only one protocol (RFC003) for trade execution at the moment.
+In the future there will be more execution protocols.
+
+The negotiation protocol should be designed in a way that it is independent from a possible orderbook.
+It should be possible to use a central or decentral orderbook for the protocol.
+
+==== Offers Constraints
+
+During the negotiation phase the offer can be changed multiple times.
+Once the negotiation phase comes to a successful end (the parties agree on the parameters for the execution) the execution phase is triggered (immediately).
+Note that there might still be an implicit accept of the message sent from taker to maker with the needed parameters.
+
+=== Definitions
+
+==== Trade ====
+
+In the COMIT context a trade is always between two digital assets.
+
+We identify two main phases when it comes to a trade:
+
+. Negotiation: The phase where the maker comes up with one or multiple offers.
+The outcome of the negotiation phase should be the parameters needed for execution.
+. Execution: The phase where the trade is executed, hence actions are executed on the ledger.
+
+Basic idea of the steps of a trade as defined by this spike:
+
+. Trade Invitation (maker -> world)
+. Negotiation (P2P)
+.. Offer Request (taker -> maker)
+.. Offer (maker -> taker)
+. Execution (P2P)
+
+==== Negotiation and Execution Roles ====
+
+The negotiation phase of a trade specifies two roles:
+
+* Maker: Creates offers
+* Taker: Takes offers
+
+Note that these roles are orthogonal to the cryptographic roles (Alice and Bob) in the execution protocol.
+Both the maker and the taker can theoretically be in the role of Alice (who generates the secret) or Bob.
+During the negotiation phase both parties have to reach agreement on who will be in the role of Alice and who will be in the role of Bob.
+
+==== Bid and Ask ====
```

Comment:
Suggest using the same terminology as trading forex.  'Trading pair', 'base asset', and 'quote asset'

ref: https://en.wikipedia.org/wiki/Currency_pair